addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4454,10 +4463,154 @@ registry.ImageOptimize = ImageHandlerOption.extend({
         return this._super(...arguments);
     },
 
+    /**
+     * @see this.selectClass for parameters
+     */
+    async shape(previewMode, widgetValue, params) {
+        if (previewMode === undefined) {
+            return;
+        }
+        await this._applyShape(widgetValue, previewMode);
+        this._getImg().classList.add('o_modified_image_to_save');
```

Comment:
This is a little bit hard to track as the modifications aren't only handled by the shape. The filter option has the same behaviour. If a filter is chosen then removed, a new attachement will still be created. Maybe it would be interesting to delete the previous attachement when a new one is created or maybe that's already done ? (server side)

Another solution would be to handle a sort of state by the option so that we know if the first modification happened during this edition session and then if all the parameters are removed we removed the tag. That's just a suggestion tho.

Reply:
We can't delete the previous attachment because it might be used elsewhere (eg if the section was duplicated or the snippet was saved). But it's true that this is a more general comment since it also applies to the filter/quality it's out of scope for this task.